packages/react-native-bridge/android/react-native-bridge/src/main/java/org/wordpress/mobile/ReactNativeGutenbergBridge/GutenbergEmbedWebViewActivity.java

```diff
@@ -0,0 +1,173 @@
+
+        mWebView = findViewById(R.id.embed_web_view);
+
+        mProgressBar = findViewById(R.id.progress_bar);
+
+        // Set settings
+        WebSettings settings = mWebView.getSettings();
+        settings.setJavaScriptEnabled(true);
+        //CookieManager cookieManager = CookieManager.getInstance();
+        //cookieManager.setAcceptThirdPartyCookies(mWebView, true);
+
+        // Setup WebView client
+        setupWebViewClient();
+
+        // Setup Web Chrome client
+        mWebView.setWebChromeClient(new WebChromeClient() {
+            @Override
+            public void onProgressChanged(WebView view, int progress) {
+                if (progress == 100) {
+                    mWebPageLoadedHandler.removeCallbacks(mWebPageLoadedRunnable);
+                    mWebPageLoadedHandler.postDelayed(mWebPageLoadedRunnable, 1500);
+                } else {
+                    mIsWebPageLoaded.compareAndSet(true, false);
+                    if (mProgressBar.getVisibility() == View.GONE) {
+                        mProgressBar.setVisibility(View.VISIBLE);
+                    }
+                    mProgressBar.setProgress(progress);
+                }
+            }
+        });
+
+        load();
+    }
+
+    protected void load() {
+        mWebView.loadUrl("https://wordpress.org/gutenberg/");
+    }
+
+    private void setupToolbar() {
+        setTitle("");
+
+        Toolbar toolbar = findViewById(R.id.toolbar);
+        if (toolbar != null) {
+            setSupportActionBar(toolbar);
+
+            ActionBar actionBar = getSupportActionBar();
+            if (actionBar != null) {
+                actionBar.setDisplayShowTitleEnabled(true);
+                actionBar.setDisplayHomeAsUpEnabled(true);
+                actionBar.setHomeAsUpIndicator(R.drawable.ic_close_24px);
+                actionBar.setSubtitle("");
+                actionBar.setTitle(getToolbarTitle());
+            }
+        }
+    }
+
+    protected String getToolbarTitle() {
+        return "";
+    }
+
+    @Override
+    public boolean onCreateOptionsMenu(Menu menu) {
+        super.onCreateOptionsMenu(menu);
+
+        MenuInflater inflater = getMenuInflater();
+        inflater.inflate(R.menu.menu_gutenberg_embed_webview, menu);
+        return true;
+    }
+
+    @Override
+    public boolean onOptionsItemSelected(final MenuItem item) {
+        if (mWebView == null) {
+            return false;
+        }
+
+        int itemID = item.getItemId();
+
+        if (itemID == android.R.id.home) {
+            finish();
+        }
+
+        return super.onOptionsItemSelected(item);
+    }
+
+    private void setupWebViewClient() {
+        mWebView.setWebViewClient(new WebViewClient() {
+            @Override
+            public void onPageStarted(WebView view, String url, Bitmap favicon) {
+                // Center the embed with a black background;
+                String css = "body{margin:0;background:#000;display:flex;align-items:center;}";
```

Comment:
The fullscreen preview on iOS has a black background by default so there is no need to check the user theme preference .